src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
```

Comment:
PullProofAI

Consider adding error handling for potential exceptions that may occur during the execution of the `catch` method.
```diff
+ try {
  private logger = new Logger();

  catch(exception: HttpException, host: ArgumentsHost) {
+ } catch (error) {
+   // Handle any potential errors here
+ }
```



<!-- This is an auto-generated comment by PullProofAI -->